backend/src/controllers/attachments.ts

```diff
@@ -152,7 +152,7 @@ export const getAttachmentController = async (req: Request, res: Response) => {
     .collection(attachmentModelConfig.attachmentCollectionName)
     .findOne({_id: new ObjectID(id)});
 
-  if (!attachment) {
+  if (!attachment || !attachment[type]) {
```

Comment:
Deze lijkt me vreemd?
Waarom deze OR?

Reply:
Dit heb ik uitgebreid specifiek voor de situatie met oudere invoices waarvoor er geen xml is opgeslagen, maar wel een pdf. Hier zal attachment niet null zijn en zal die een error smijten in de lijn nadien omdat er voor type xml geen buffer beschikbaar is.